core/src/main/java/org/infinispan/expiration/impl/TxClusterExpirationManager.java

```diff
@@ -0,0 +1,15 @@
+package org.infinispan.expiration.impl;
+
+import java.util.concurrent.CompletableFuture;
+
+import net.jcip.annotations.ThreadSafe;
+
+@ThreadSafe
+public class TxClusterExpirationManager<K, V> extends ClusterExpirationManager<K, V> {
+   @Override
+   CompletableFuture<Boolean> handleMaxIdleExpireEntry(K key, V value, long maxIdle) {
+      // We cannot remove an entry in a tx, due to possible dead lock - thus we just return
+      // if the entry is expired or not
```

Comment:
Pessimistic cache any write operation and read with `FORCE_WRITE_LOCK` would cause the issue as it first locks the entry then it would fire the remove expired on a separate thread that it waits for.

Reply:
Sure, but `TxClusterExpirationManager` is used for all tx caches, not just pessimistic ones. Reading this comment it feels like max-idle expiration is basically disabled in all tx caches.

Reply:
That is why it is documented in user guide that max idle in a tx cache only removes entries via reaper thread. We can obviously improve this for optimistic tx, if need be. But I was trying to get it working sooner rather than later :)

Reply:
I created https://issues.jboss.org/browse/ISPN-9271 to change this later if needed.